src/library_html5.js

```diff
@@ -929,23 +929,38 @@ var LibraryHTML5 = {
   },
 
   $screenOrientation: () => {
-    if (!screen) return undefined;
-    return screen.orientation || screen.mozOrientation || screen.webkitOrientation || screen.msOrientation;
+    if (!window.screen) return undefined;
+    return screen.orientation || screen['mozOrientation'] || screen['webkitOrientation'];
   },
 
   $fillOrientationChangeEventData__deps: ['$screenOrientation'],
   $fillOrientationChangeEventData: (eventStruct) => {
-    var orientations  = ["portrait-primary", "portrait-secondary", "landscape-primary", "landscape-secondary"];
-    var orientations2 = ["portrait",         "portrait",           "landscape",         "landscape"];
-
-    var orientationString = screenOrientation();
-    var orientation = orientations.indexOf(orientationString);
-    if (orientation == -1) {
-      orientation = orientations2.indexOf(orientationString);
+    // OrientationType enum
+    var orientationsType1 = ['portrait-primary', 'portrait-secondary', 'landscape-primary', 'landscape-secondary'];
+    // alternative selection from OrientationLockType enum
+    var orientationsType2 = ['portrait',         'portrait',           'landscape',         'landscape'];
+
+    var orientationIndex = 0;
+    var orientationAngle = 0;
+    var screenOrientObj  = screenOrientation();
+    if (typeof screenOrientObj === 'object') {
+      orientationIndex = orientationsType1.indexOf(screenOrientObj.type);
+      if (orientationIndex < 0) {
+        orientationIndex = orientationsType2.indexOf(screenOrientObj.type);
+      }
+      if (orientationIndex >= 0) {
+        orientationIndex = 1 << orientationIndex;
+      }
+      orientationAngle = screenOrientObj.angle;
+#if MIN_SAFARI_VERSION < 0x100400
+    } else {
+      // fallback for Safari earlier than 16.4 (March 2023)
+      orientationAngle = window.orientation;
+#endif
```

Comment:
Sorry can you move the if/else here so they come after the closing brace for the `if`?   Sorry for being picky.

Reply:
e.g.
 
```
if (xx) {
}
#if XXX
else {
}
#endif
```

Reply:
Sure, NP.